huaweicloud/services/coc/data_source_huaweicloud_coc_war_rooms.go

```diff
@@ -0,0 +1,415 @@
+		}
+		if currentOffset == 0 {
+			mErr = multierror.Append(
+				d.Set("running_num", utils.PathSearch("data.running_num", listRespBody, nil)),
+				d.Set("closed_num", utils.PathSearch("data.closed_num", listRespBody, nil)),
+				d.Set("total_num", utils.PathSearch("data.total_num", listRespBody, nil)),
+			)
+		}
+		listWarRooms := flattenCocListWarRooms(listRespBody)
+		if len(listWarRooms) < 1 {
+			break
+		}
+		list = append(list, listWarRooms...)
+		currentOffset += len(listWarRooms)
+	}
+
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("list", list),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildListWarRoomsBodyParams(d *schema.ResourceData, currentOffset int) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"limit":                    100,
+		"offset":                   currentOffset,
+		"admin":                    utils.ValueIgnoreEmpty(d.Get("admin")),
+		"current_users":            utils.ValueIgnoreEmpty(d.Get("current_users")),
+		"enterprise_project_ids":   utils.ValueIgnoreEmpty(d.Get("enterprise_project_ids")),
+		"impacted_application_ids": utils.ValueIgnoreEmpty(d.Get("impacted_application_ids")),
+		"incident_levels":          utils.ValueIgnoreEmpty(d.Get("incident_levels")),
+		"incident_num":             utils.ValueIgnoreEmpty(d.Get("incident_num")),
+		"notification_level":       utils.ValueIgnoreEmpty(d.Get("notification_level")),
+		"occur_end_time":           utils.ValueIgnoreEmpty(d.Get("occur_end_time")),
+		"occur_start_time":         utils.ValueIgnoreEmpty(d.Get("occur_start_time")),
+		"recover_end_time":         utils.ValueIgnoreEmpty(d.Get("recover_end_time")),
+		"recover_start_time":       utils.ValueIgnoreEmpty(d.Get("recover_start_time")),
+		"region_code_list":         utils.ValueIgnoreEmpty(d.Get("region_code_list")),
+		"statistic_flag":           utils.ValueIgnoreEmpty(d.Get("statistic_flag")),
+		"status":                   utils.ValueIgnoreEmpty(d.Get("status")),
+		"title":                    utils.ValueIgnoreEmpty(d.Get("title")),
+		"triggered_end_time":       utils.ValueIgnoreEmpty(d.Get("triggered_end_time")),
+		"triggered_start_time":     utils.ValueIgnoreEmpty(d.Get("triggered_start_time")),
+		"war_room_num":             utils.ValueIgnoreEmpty(d.Get("war_room_num")),
+		"war_room_nums":            utils.ValueIgnoreEmpty(d.Get("war_room_nums")),
+	}
+	return bodyParams
+}
+
+func flattenCocListWarRooms(resp interface{}) []map[string]interface{} {
+	warRoomsJson := utils.PathSearch("data.list", resp, make([]interface{}, 0))
+	warRoomsArray := warRoomsJson.([]interface{})
+	if len(warRoomsArray) == 0 {
+		return nil
+	}
+
+	result := make([]map[string]interface{}, 0, len(warRoomsArray))
+	for _, warRoom := range warRoomsArray {
+		result = append(result, utils.RemoveNil(map[string]interface{}{
```

Comment:
no need to add utils.RemoveNil()
```suggestion
		result = append(result, utils.RemoveNil(map[string]interface{}{
```